### PR TITLE
KYLIN-3514 Support AUC UDAF

### PR DESCRIPTION
**Pull request by r7raul1984**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: core-metadata/pom.xml [hunk 1]
```diff
@@ -24,6 +24,18 @@
     <packaging>jar</packaging>
     <name>Apache Kylin - Core Metadata</name>
     <description>Apache Kylin - Core Metadata</description>
+    <build>
```
Comment (yiming187) [inline core-metadata/pom.xml]: Do you mean this commit only works in JDK8+?

Comment (r7raul1984) [reply core-metadata/pom.xml]: I use stream api in my code. And I found UT on master branch with JDK 1.7(deprecated): Kylin-Master-JDK-1.7  in kylin docs.So I add this config.

File: core-metadata/pom.xml [hunk 2]
```diff
@@ -87,6 +99,10 @@
             <groupId>com.tdunning</groupId>
             <artifactId>t-digest</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.github.haifengl</groupId>
+            <artifactId>smile-core</artifactId>
```
Comment (shaofengshi) [inline core-metadata/pom.xml]: Could you please add the copyright of smile-core to Kylin's LICENSE file in the root folder?

**Comment by shaofengshi** [comment]
How to use this function in Kylin? I didn't see the UDF implementation. Please provide some description and sample for us to understand this function. Thanks!

**Comment by shaofengshi** [comment]
This is not a complete implementation, for example, it doesn't translate the query to this measure type. Will close it now, and revisit it later if find a total solution. Thank you!

**Pull request closed without merging**